contracts/l2/FuseController.sol

```diff
@@ -0,0 +1,309 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.17;
+
+import "./L2Registry.sol";
+import "./IFuseController.sol";
+import "./IControllerUpgrade.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+error Unauthorised(bytes32 node, address addr);
+error CannotUpgrade();
+error nameExpired(bytes32 node);
+
+/**
+ * @dev A simple ENS registry controller. Names are permanently owned by a single account.
+ *      Name data is structured as follows:
+ *       - Byte 0: controller (address)
+ *       - Byte 20: owner (address)
+ *       - Byte 40: resolver (address)
+ *       _ Byte 60: expiry (uint64)
+ *       - Byte 68: fuses (uint96)
```

Comment:
If you reduce this to a `uint64`, the whole thing will be 96 bytes - saving a storage slot.